### ISSUE TITLE
Can't find Documentation on retention parameters

### ISSUE THREAD

**Issue by taschaal** [ansible-collections/netapp.ontap#144]
Error creating volume xxxxxxx: calling: storage/volumes: got {'message': 'The duration value \"p3h\" is invalid. Expected input is (<\"Duration in the ISO 8601 format | infinite | unspecified\">)', 'code': '918253', 'target': 'snaplock.retention.maximum'}.",

I've tried p3h, P3H,  3h, 3H, '3hours'.  Nothing seems to work.  It would be helpful if the documentation included an example.

**Comment by mohanrajb9** [comment]
retention doc points autocommit_period for duration format.
https://docs.ansible.com/ansible/latest/collections/netapp/ontap/na_ontap_volume_module.html#parameter-snaplock/autocommit_period
examples are P30M, P10Y, PT1H, “none”. A duration that combines different periods is not supported.

Filters - iso8601_duration_to_seconds, iso8601_duration_from_seconds are added in 21.24.0, later filter can be used to convert seconds to ISO format.

```
- name: volume task.
      na_ontap_volume:
        state: present
        name: "{{ snaplock_volume }}"
        aggregate_name: "{{ aggregate }}"
        size: 20
        size_unit: mb
        space_guarantee: none
        policy: default
        type: rw
        snaplock:
          type: enterprise
          retention:
            default: "{{ 60 | netapp.ontap.iso8601_duration_from_seconds }}"
```

we will add this in volume module examples.

**Comment by carchi8py** [comment]
This was fixed as part of 22.0.0
